### PR TITLE
Refactor `test_deps_compat`

### PR DESCRIPTION
**Pull request by lgoettgens**
In preparation for fixing https://github.com/JuliaTesting/Aqua.jl/issues/178.

Changes existing functions to return intermediate results instead of `LazyTestResult`s. This enables better composability without repeating too much work. Furthermore, the internal tests got clearer and more readable.

### CODE REVIEW

File: src/deps_compat.jl [hunk 1]
```diff
@@ -1,84 +1,73 @@
...
 - `ignore::Vector{Symbol}`: names of dependent packages to be ignored.
 """
-function test_deps_compat(packages; kwargs...)
-    @testset "$(result.label)" for result in analyze_deps_compat(packages; kwargs...)
-        @debug result.label result
-        @test result ⊜ true
+function test_deps_compat(pkg::PkgId; broken::Bool = false, kwargs...)
+    result = find_missing_deps_compat(pkg; kwargs...)
+    if broken
+        @test_broken result == []
```
Comment (fingolfin) [inline src/deps_compat.jl:18]: Maybe do this here and in multiple other places? It avoids allocating a temporary vector. Then again performance doesn't really matter here, so... *shrug*
```suggestion
        @test_broken isempty(result)
```

**Review by fingolfin — APPROVED** [review]

File: src/deps_compat.jl (hunk 1 above)
Comment (lgoettgens) [reply src/deps_compat.jl:18]: I'll do that in this PR for this tests and something analogous for the rest in a future PR.

File: src/deps_compat.jl [hunk 2]
```diff
@@ -1,84 +1,73 @@
...
+    else
+        @test result == []
```
Comment (lgoettgens) [inline src/deps_compat.jl:20]:
```suggestion
        @test isempty(result)
```

File: test/test_deps_compat.jl [hunk 3]
```diff
@@ -1,82 +1,73 @@
...
-            root_project_path,
+        result = find_missing_deps_compat(
             DictSA("deps" => DictSA(), "compat" => DictSA("julia" => "1")),
-        ) ⊜ true
-        @test _analyze_deps_compat_2(
-            pkg,
-            root_project_path,
+            "deps",
+        )
+        @test result == []
```
Comment (lgoettgens) [inline test/test_deps_compat.jl:14]:
```suggestion
        @test isempty(result)
```

File: test/test_deps_compat.jl [hunk 4]
```diff
@@ -1,82 +1,73 @@
...
+        result = find_missing_deps_compat(
             DictSA(
                 "deps" => DictSA("PkgA" => "229717a1-0d13-4dfb-ba8f-049672e31205"),
                 "compat" => DictSA("julia" => "1", "PkgA" => "1.0"),
             ),
-        ) ⊜ true
+            "deps",
+        )
+        @test result == []
```
Comment (lgoettgens) [inline test/test_deps_compat.jl:30]:
```suggestion
        @test isempty(result)
```

File: test/test_deps_compat.jl [hunk 5]
```diff
@@ -1,82 +1,73 @@
...
         @testset "does not have `deps`" begin
-            # Not sure if it should fail or passs:
-            t = _analyze_deps_compat_2(pkg, root_project_path, DictSA())
-            @test t ⊜ true
-            @test occursin("does not have `deps`", string(t))
+            result = find_missing_deps_compat(DictSA(), "deps")
+            @test result == []
```
Comment (lgoettgens) [inline test/test_deps_compat.jl:33]:
```suggestion
        @test isempty(result)
```

**Pull request merged**